File: mobile/lib/models/achievement.dart
class Achievement {
  int id;
  String name;
  String description;
  String icon;

  Achievement({
    required this.id,
    required this.name,
    required this.description,
    required this.icon,
  });

  factory Achievement.fromJson(Map<String, dynamic> json) => Achievement(
        id: json["id"],
        name: json["name"],
        description: json["description"],
        icon: json["icon"],
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "name": name,
        "description": description,
        "icon": icon,
      };
}

File: backend/api/Models/Achievement.cs
using api.DTO;
using System.Text.Json.Serialization;

namespace api.Models;

public class Achievement
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Icon { get; set; }
    [JsonIgnore]
    public List<User> Users { get; set; } = [];

    public Achievement Update(AchievementUpdateRequest request)
    {
        Name = request.Name;
        Description = request.Description;
        Icon = request.Icon;
        return this;
    }
}
